homeassistant/components/fido/sensor.py

```diff
@@ -33,27 +37,108 @@
 REQUESTS_TIMEOUT = 15
 MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=15)
 
+
 SENSOR_TYPES = {
-    "fido_dollar": ["Fido dollar", PRICE, "mdi:cash-usd"],
-    "balance": ["Balance", PRICE, "mdi:cash-usd"],
-    "data_used": ["Data used", DATA_KILOBITS, "mdi:download"],
-    "data_limit": ["Data limit", DATA_KILOBITS, "mdi:download"],
-    "data_remaining": ["Data remaining", DATA_KILOBITS, "mdi:download"],
-    "text_used": ["Text used", MESSAGES, "mdi:message-text"],
-    "text_limit": ["Text limit", MESSAGES, "mdi:message-text"],
-    "text_remaining": ["Text remaining", MESSAGES, "mdi:message-text"],
-    "mms_used": ["MMS used", MESSAGES, "mdi:message-image"],
-    "mms_limit": ["MMS limit", MESSAGES, "mdi:message-image"],
-    "mms_remaining": ["MMS remaining", MESSAGES, "mdi:message-image"],
-    "text_int_used": ["International text used", MESSAGES, "mdi:message-alert"],
-    "text_int_limit": ["International text limit", MESSAGES, "mdi:message-alert"],
-    "text_int_remaining": ["International remaining", MESSAGES, "mdi:message-alert"],
-    "talk_used": ["Talk used", TIME_MINUTES, "mdi:cellphone"],
-    "talk_limit": ["Talk limit", TIME_MINUTES, "mdi:cellphone"],
-    "talk_remaining": ["Talk remaining", TIME_MINUTES, "mdi:cellphone"],
-    "other_talk_used": ["Other Talk used", TIME_MINUTES, "mdi:cellphone"],
-    "other_talk_limit": ["Other Talk limit", TIME_MINUTES, "mdi:cellphone"],
-    "other_talk_remaining": ["Other Talk remaining", TIME_MINUTES, "mdi:cellphone"],
+    "fido_dollar": SensorEntityDescription(
+        key="Fido dollar",
+        unit_of_measurement=PRICE,
+        icon="mdi:cash-usd",
+    ),
```

Comment:
```suggestion
    SensorEntityDescription(
        key="fido_dollar",
        name="Fido dollar",
        unit_of_measurement=PRICE,
        icon="mdi:cash-usd",
    ),
```

This applies to the other entries as well

Reply:
but it doubles the name/key text constant, and is not really needed. I see the problem more that you choose to have "key" as demanded, where NamedTuple uses "name".

Reply:
It doesn't double anything? Move this from a dict into a tuple, moving the current dict key into the description class.